### PR TITLE
TCM-284 | Allow user to register and login

### PR DESCRIPTION
**Pull request by VivekV95**
## Description
Setup up authentication flows for user login and registration

### Summary

- [x] Allow user to register an account
- [x] Allow user to sign in 

### Issue

[XPP-284](https://hyfnla.atlassian.net/browse/TCM-284)

### Video/Screenshot

[Login success](https://drive.google.com/file/d/1-39XefdngyGkDxNmUVz6FHSKvL0W3W4h/view?usp=sharing)
[Register confirm password validation](https://drive.google.com/file/d/1eljPM3LT9dMBi8ccaWpJyXJi2N2WVWv5/view?usp=sharing)
[Register email validation](https://drive.google.com/file/d/1wF14gnnmDGdcM8g1skeP0CQtnVEpeHxI/view?usp=sharing)
[Register password validation](https://drive.google.com/file/d/1rNiHT72o3a_5-y6Sn5FARAtRC8WHvFos/view?usp=sharing)
[Register success](https://drive.google.com/file/d/1LkWOc42ZzITWsXtXUR_IO_e1sIOpPCeT/view?usp=sharing)

### CODE REVIEW

**Comment by VivekV95** [comment]
@onodude Will taskify the todos in this and move on to those next if that's fine

File: app/src/main/java/com/trx/consumer/base/BaseApi.kt [hunk 1]
```diff
@@ -17,35 +17,35 @@ interface BaseApi {
...
-        @Body params: HashMap<String, Any>
+        @Body params: Map<String, @JvmSuppressWildcards Any>
     ): Response<Any>
 
     // Todo: May need to use @HTTP(method = "PATCH"), needs investigation.
     @PATCH
     suspend fun patch(
         @Url url: String,
         @Header("Authorization") authorization: String? = null,
-        @Body params: HashMap<String, Any>
```
Comment (onodude) [inline app/src/main/java/com/trx/consumer/base/BaseApi.kt:49]: why was this change necessary?

Comment (VivekV95) [reply app/src/main/java/com/trx/consumer/base/BaseApi.kt]: Reverted [here](https://github.com/hyfn/trx-consumer-android/pull/29/commits/13cf5952403a833f7c058592b10d239c3b78d8ac#diff-414e28a5266a6b861c2877802ba887d103ed6f567c86f30f343f5532485d25f3R20)

**Review by onodude — APPROVED** [review]
